src/js/steps/getData.js

```diff
@@ -10,13 +10,76 @@ var $ = require('jquery');
  *                   information on the pages. Seriously, just use a debugger.
  */
 module.exports = function (options) {
-  return $.getJSON(options.server, { book: options.book })
-    .then(function (data) {
-      // For backwards compatibility
-      if (!data.book.spreads) {
-        data.book.spreads = 'double';
-      }
+  var returnPromise;
+  if (!options.platformAPI) {
+    returnPromise = $.getJSON(options.server, { book: options.book })
+      .then(function (data) {
+        // For backwards compatibility
+        if (!data.book.spreads) {
+          data.book.spreads = 'double';
+        }
 
-      return data.book;
-    });
+        return data.book;
+      });
+  } else {
+    console.log('options.server', options.server);
+    returnPromise = $.ajax({
+      dataType: 'JSON',
+      url: options.server + 'product-builder/tjh/images',
+      data: options.book
+    })
+      .then(function (data) {
+        var transformedData = {
+          book: {
+            spreads: 'single',
+            letters: generatePlatformURLs(data)
+          }
+        };
+        return transformedData.book;
+      });
+  }
+  return returnPromise;
 };
+
+function generatePlatformURLs(data) {
+  var dynamicPages = data.images.map(function (image) {
+    return {
+      url: image.url,
+      id: image.id,
+      type: 'story'
+    };
+  });
+
+  var frontPages = [
+    {
+      url: 'https://s3-eu-west-1.amazonaws.com/tjh-preview-assets/TheJourneyHome_FrontCover_RGB.jpg',
```

Comment:
Yeah, that’s a better plan.

> On 28 Apr 2016, at 11:24, Patrick Sinclair <notifications@github.com> wrote:
> 
> In src/js/steps/getData.js <https://github.com/Lostmyname/monkey/pull/73#discussion_r61404966>:
> 
> >  };
> > +
> > +function generatePlatformURLs(data) {
> > +  var dynamicPages = data.images.map(function (image) {
> > +    return {
> > +      url: image.url,
> > +      id: image.id,
> > +      type: 'story'
> > +    };
> > +  });
> > +
> > +  var frontPages = [
> > +    {
> > +      url: 'https://s3-eu-west-1.amazonaws.com/tjh-preview-assets/TheJourneyHome_FrontCover_RGB.jpg',
> Wouldn't it be best to drive these through imgix so we can scale them?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/Lostmyname/monkey/pull/73/files/d8c1bec533c3ef8ea7e2d6de48f405f716bebaa1#r61404966>